front/app/hooks/useTopics.ts

```diff
@@ -4,24 +4,36 @@ import {
   topicByIdStream,
   topicsStream,
   Code,
+  ITopicsQueryParams,
 } from 'services/topics';
 import { Observable, of, combineLatest } from 'rxjs';
 import { map } from 'rxjs/operators';
 import { isNilOrError, NilOrError, reduceErrors } from 'utils/helperUtils';
 
-interface Parameters {
+export interface Parameters {
+  /** Don't use the ids and the query parameters (code, exclude_code, sort) together.
+   *  Only one of the two at a time.
+   */
   topicIds?: string[];
   code?: Code;
-  exclude_code?: Code;
+  excludeCode?: Code;
   sort?: 'new' | 'custom';
+  forHomepageFilter?: boolean;
 }
 
-export default function useTopics(parameters: Parameters) {
-  const { topicIds, code, exclude_code, sort } = parameters;
+export default function useTopics(parameters: Parameters = {}) {
+  const { topicIds, code, excludeCode, sort, forHomepageFilter } = parameters;
   const [topics, setTopics] = useState<ITopicData[] | NilOrError>(undefined);
-  const queryParameters = { code, exclude_code, sort };
 
   const topicIdsStringified = JSON.stringify(topicIds);
+
+  const queryParameters: ITopicsQueryParams = {
+    code,
+    exclude_code: excludeCode,
+    sort,
+    for_homepage_filter: forHomepageFilter,
```

Comment:
Just curious - what does it mean that is for homepage filter?

Reply:
If `for_homepage_filter=true`, the response contains only topics that are used for at least one project. See https://developers.citizenlab.co/api-docs/ee/frontweb_api/master/topics/list_only_selected_topics_does_not_include_topics_only_used_by_draft_projects.html. There is an analogous parameter for `areas`.